airbyte-cdk/python/airbyte_cdk/connector_builder/message_grouper.py

```diff
@@ -222,10 +223,12 @@ def _get_message_groups(
                 datetime_format_inferrer.accumulate(message.record)
             elif message.type == MessageType.CONTROL and message.control.type == OrchestratorType.CONNECTOR_CONFIG:
                 yield message.control
+            elif message.type == MessageType.STATE:
+                latest_state_message = message.state
```

Comment:
@girarda my understanding of this feature was that every slice would output its own state message, so we would expect every item in the `slices` response of the `read_stream` request to contain a `state` object.

But when I tested this, only the last slice had a `state` in the response of `read_stream`.

Is this expected that only the last slice will ever output a state message? If so, why does this happen instead of each slice outputting a state?

Reply:
how did you test this? My initial hypothesis is this might be because we don't necessarily produce a stream slice due to record and page limits but I can dig more into the issue

Reply:
I used the exchange rates connector that you build in the tutorial:
[exchange_rates.yaml](https://github.com/airbytehq/airbyte/files/14999768/exchange_rates.yaml.txt)

And I pointed my local connector builder server at the local CDK on this branch

Reply:
The issue is that the builder server submits a request for a full refresh read instead of incremental. Fix is here https://github.com/airbytehq/airbyte-platform-internal/pull/12179